File: csharp/hamming/Hamming.cs
using System;
using System.Linq;

public static class Hamming
{
    public static int Distance(string firstStrand, string secondStrand) => isEqualLength(firstStrand, secondStrand) ? _Distance(firstStrand, secondStrand) : throw new ArgumentException();

    private static int _Distance(string firstStrand, string secondStrand) => firstStrand.Zip(secondStrand, (firstChar, secondChar) => firstChar == secondChar).Where(x => x == false).Count();

    private static bool isEqualLength(string firstStrand, string secondStrand) => firstStrand.Length == secondStrand.Length;
}

File: haskell/hamming/src/Hamming.hs
module Hamming (distance) where

distance :: String -> String -> Maybe Int
distance xs ys
  | equalLength xs ys = Just (distance' xs ys)
  | otherwise = Nothing

distance' :: String -> String -> Int
distance' xs ys = length $ filter id $ zipWith (/=) xs ys

equalLength :: String -> String -> Bool
equalLength xs ys = length xs == length ys